ADD l10n_es_facturae

JuanjoA opened the pull request:
Añadido el módulo de facturación electrónica para la versión 7.0 de
openerp.
Emite el XML que se enviará a la administración.
Pendiente de test con datos reales.

Commit:
Mejora general módulo city (bug, textos, limpieza datos). Mejoras importación datos externos en l10n_ES_toponyms, l10n_ES_toponyms_CA, l10n_CT_toponyms

Commit:
Módulo l10n_ES_extractos_bancarios: Lectura ficheros codificados en UTF-8 e ISO-8859-1

Commit:
Módulo l10n_ES_extractos_bancarios: Lectura ficheros codificados en UTF-8 e ISO-8859-1

Commit:
Módulo l10n_ES_remesas: Eliminación espacios de más en el sangrado en remesas.py

Commit:
Módulo l10n_ES_remesas: Eliminación espacios de más en el sangrado en remesas.py

acysos committed:
Fiscal year close routines added

albertca committed:
Fixed typo in serveral places: "Todas los asientos" -> "Todos los asientos".

Commit:
quito extensión py de import

Commit:
Corrección identaciones módulo stock_past

albertca committed:
Remove the default 'To be installed' flag in account_invoice_series and sale_paytype modules.

albertca committed:
Añadido un pequeño script para simplificar la creación de enlaces simbolicos a los módulos en el servidor.

albertca committed:
Merge.

Commit:
borrado sale_autopicking, en desarrollo

Commit:
eliminado sale_autopicking. en desarrollo

albertca committed:
Añadidas comprobaciones para asegurarnos que existe un partner y un producto
en la formula de decisión de si se aplica el recargo de equivalencia. Si no
se verifica puede causar errores en algunas llamadas a la funcion compute()
de 'account.tax'.

albertca committed:
Añadida la traducción de un widget.

albertca committed:
Remove a couple of bytecode files.

albertca committed:
Merge with removal of sale_autopicking module.

Commit:
Actualización rml informe de remesas de recibos

salbet committed:
Corregido error de codificación con el nombre de la compañía

Commit:
Importación extractos bancarios: Comprobación previa que el extracto no esté confirmado

Commit:
Nuevo módulo c2c_finance_report: Mejoras informes financieros traducidos al castellano

Commit:
Correcciones módulo c2c_finance_report: Traducciones al castellano, formato fecha DD/MM/AA, líneas erròneas en el mayor

Commit:
l10n_ES_account_report: Correcciones y mejoras (antiguo módulo account_ES_reporting), l10n_chart_ES: Eliminación informes vacios, c2c_finance_report: Pequeñas mejoras y traducciones

Commit:
l10n_ES_account_report: Pequeñas correcciones antes de copiarlo a la rama 5.0

Commit:
l10n_chart_ES: Pequeñas mejoras y corrección amplia del asistente de cierre ejercicio fiscal

Commit:
[FIX] l10n_ES_remesas: Creación asientos remesas con 'amount_currency': 0. Conciliación asientos remesas con cuenta like '67800%0' y diario ilike '%general%' para el posible desfase

rlizana committed:
[MIG] Eliminado bzrignore y creado gitignore

pedrobaeza committed:
Merge pull request #17 from rlizana/4.2

[MIG] Eliminado bzrignore y creado gitignore

Commit:
ADD l10n_es_facturae

Añadido el módulo de facturación electrónica para la versión 7.0 de
openerp.
Emite el XML que se enviará a la administración.
Pendiente de test con datos reales.

pedrobaeza commented on l10n_es_facturae/__init__.py:
```diff
@@ -0,0 +1,27 @@
+#
+##############################################################################
+
+import country
```
Utiliza `from . import ...`

pedrobaeza commented on l10n_es_facturae/__openerp__.py:
```diff
@@ -0,0 +1,43 @@
+    "author" : "Alejandro Sanchez",
+    "category" : "Localisation/Accounting",
+    "description" : """This module creates Factura-e 3.1 files from invoices.
+3.1 -> 3.2 converter: http://sedeaplicaciones2.minetur.gob.es/FacturaECon/
```
Ponlo en español puesto que es un módulo español y > 80 cols

pedrobaeza commented on l10n_es_facturae/__openerp__.py:
```diff
@@ -0,0 +1,43 @@
+    "website" : "www.asr-oss.com",
+    "license" : "AGPL-3",
+    "depends" : ["base", "account"],
+    "update_xml" : [
```
'data'

pedrobaeza commented on l10n_es_facturae/country.py:
```diff
@@ -0,0 +1,55 @@
+##############################################################################
+#
+#    OpenERP, Open Source Management Solution
+#    Copyright (c) 2009 Alejandro Sanchez (http://www.asr-oss.com) All Rights Reserved.
```
Quitar All Rights Reserved. Eso no aplica desde hace muchos años.

pedrobaeza commented on l10n_es_facturae/country.py:
```diff
@@ -0,0 +1,55 @@
+#    OpenERP, Open Source Management Solution
+#    Copyright (c) 2009 Alejandro Sanchez (http://www.asr-oss.com) All Rights Reserved.
+#                       Alejandro Sanchez <alejandro@asr-oss.com>
+#    $Id$
```
Líneas inútil

pedrobaeza commented on l10n_es_facturae/country.py:
```diff
@@ -0,0 +1,55 @@
+#
+#    OpenERP, Open Source Management Solution
+#    Copyright (c) 2009 Alejandro Sanchez (http://www.asr-oss.com) All Rights Reserved.
+#                       Alejandro Sanchez <alejandro@asr-oss.com>
```
Línea duplicada

pedrobaeza commented on l10n_es_facturae/country.py:
```diff
@@ -0,0 +1,55 @@
+    }
+
+    def _auto_init(self, cr, context=None):
+        res = super(Country, self)._auto_init(cr, context=context)
```
Mira lo que he puesto de comentario en la 6.1

pedrobaeza commented on l10n_es_facturae/country.py:
```diff
@@ -0,0 +1,55 @@
+        return res
+
+
+# vim:expandtab:smartindent:tabstop=4:softtabstop=4:shiftwidth=4:
```
Línea inútil

pedrobaeza commented:
En general, falta lo mismos comentarios que en la 6.1, y el formato PEP8.

Commit:
[IMP] PEP8 y otros

Se realizan los cambios comentados por Pedro y alguno más.
Se añade un label y un placeholder en los campos del certificado y la
contraseña para saber a qué se refieren.

JuanjoA commented:
Ok, cambios realizados y algún que otro más.
Se vuelve a testear y sigue validando el XML correctamente.

Commit:
[DEL] Eliminado fichero copia

Una copia de un fichero que se me había colado, eliminado.

Commit:
PEP8 más ajustes

Eliminados import no necesarios y alguna variable de más.
Ajustes de líneas >80

Commit:
[PEP8] Mejorando y corrección campo no accesible

Commit:
PEP8 - peleando con travis para la validación

Commit:
PEP8 - líneas

Commit:
PEP8 - pylint creo que último cambio

omar7r commented on l10n_es_facturae/wizard/create_facturae.py:
```diff
@@ -0,0 +1,687 @@
+                texto += '<TaxRate>0.00</TaxRate>'
+                texto += '<TaxableBase>'
+                texto += '<TotalAmount>' + \
+                         str('%.2f' % line.price_subtotal) + '</TotalAmount>'
```
El TotalAmount de TaxesWithheld debe ser 0

omar7r commented on l10n_es_facturae/partner_view.xml:
```diff
@@ -0,0 +1,65 @@
+                </field>
+            </field>
+        </record>
+        -->
```
Eliminar código comentado

Commit:
errores en pull

Commit:
Notas de Omar

Cambios según notas de Omar

JuanjoA commented:
He aplicado cambios comentados por @omar7r
Por otro lado, el commit de errores en pull, es por una serie de conflictos en la sincronización, no se si esto estropeará el PR, supongo que la rama está correcta, si alguién puede revisarlo estupendo.
Gracias

pedrobaeza commented:
Luego cuando hago merge, lo hago manual con un squash y limpio el PR.

pedrobaeza commented:
Ah, pero no, estoy viendo que has mezclado cosas de la rama 5.0, conflictos, etc... Haz una cosa:
- Empieza con una rama 7.0 limpia.
- Haz `git mv` del directorio de facturae que está en `__unported__` para sacarlo de ahí.
- Copia y pega tu versión en él.
- Haz un commit.
- Súbelo con `git push -f` a esta misma rama.
  Y así, se quedará todo limpio.

JuanjoA commented:
Ok, por lo que veo se me ha jodido la rama.
Lo que no se es porque se trae esos errores a mi rama 7.0 de ramas anteriores.
Pruebo ahora mismo.
Thx

Commit:
Revert "Notas de Omar"

This reverts commit 1bd29fd1e12fb3bf5304b187640a9aec467cb3e0.

Commit:
Revert "errores en pull"

This reverts commit 6eb27ca76a8be2779835566d5982910147fc5bf5, reversing
changes made to 719371d57116f318a2be1802c6696cb8fe0704cc.

Commit:
Revert 6a9a1bb..7e46b8c

This rolls back to commit 6a9a1bb2846d6e78d08c911c887f7e0521947a43.

Commit:
Revert "Revert "Notas de Omar""

This reverts commit 6a9a1bb2846d6e78d08c911c887f7e0521947a43.

Commit:
Revert "errores en pull"

This reverts commit 6eb27ca76a8be2779835566d5982910147fc5bf5, reversing
changes made to 719371d57116f318a2be1802c6696cb8fe0704cc.

Conflicts:
	l10n_es_facturae/wizard/create_facturae.py

Commit:
Revert 719371d..37a5b9d

This rolls back to commit 719371d57116f318a2be1802c6696cb8fe0704cc.

Commit:
Reparados errores commit erroneo

Rama limpia, con los cambios de Omar sobre totalamount en
create_facturae.py

Commit:
Eliminado facturae de __unported__

Commit:
Eliminados comentarios en partner

JuanjoA commented:
Ok, reparado, aunque aún no se porque se me han descargado cambios de ramas anteriores en la mía 7.0
Quizás tema de usar el sw git, en vez de línea de comandos.
Ya está ok otra vez, con los últimos cambios

ismaelcj commented:
Hola. Por tema de que una factura puediese tener dirección de factura y dirección de contacto distintos estamos mirando como se genera la factura-e, en la 6.1 y los nombres de la administración (organo gestor, unidad tramitadora...) lo estaba cogiendo del campo 'Dirección de Factura' y creemos que debería de cogerlo del campo 'Dirección de contacto' en la pestaña Otra información. En la 7.0 se está cogiendo directamente del partner, ¿se debería crear otro campo como el de 'Dirección de contacto'?

JuanjoA commented:
Ok, el estado del módulo en este momento (en versión 7):
- puedes generar el xml de la factura
- no funciona la firma con el .jar incluido en linux
- ese .xml lo puedes firmar con el programa que provee nuestro <i>amado gobierno</i>
  link: http://www.facturae.gob.es/formato/Paginas/descarga-aplicaciones.aspx
  Sólo me ha funcionado desde IE (uff, java, IE y gobierno, todo junto)
  Se firma con el certificado de la empresa
- con el fichero generado, .xsig, ya puedes subirlo a face.gob.es, con el certificado de persona física (gracias Alejandro Santana)
- te lo acepta y a correr

He detectado un error que se produce si el nombre del cliente supera los 40 caractéres, lo que habrá que controlar a la hora de ejecutar el asistente de creación desde el ERP, para avisar al usuario, ya que el error que da la plataforma es numérico (en la página de test si te suelta el problema, en la principal de remitir tus facturas, no, mola!)

p.d. tag <i> de irónico, evidentemente
p.d.2 el programa de la administración tiene guasa!

omar7r commented:
Hola @JuanjoA,

Lo acabo de migrar a la versión 8.0 https://github.com/OCA/l10n-spain/pull/214 y funciona bien en Linux la firma con el .jar igual que en la 6.1.
Creo que deberíamos añadirlo ya al repositorio, que no tiene sentido tenerlo así desde hace tanto tiempo, luego ya se le añadirán mejoras, por ejemplo lo de que no exporta correctamente las rectificativas, tal y como está para facturas de venta funciona bien y las firma correctamente.

Un saludo.

pedrobaeza commented:
Aquí también hay que hacer rebase. Documentad por favor lo que falta y una guía de configuración.

pedrobaeza commented:
He añadido consideraciones al PR de la rama v8. Revisadlas por favor. Mientras no estén cumplidas, para mí esto no tendría el visto bueno para ser mergeado en la localización.

JuanjoA commented:
Buenas, ahora mismo, estoy a tope, con poco tiempo, lo miro en cuanto pueda, por lo menos para terminar este PR, gracias a ambos.

rafaelbn commented:
Estaré encantado de revisarlo y probarlo cuando esté listo. ¡Muchas gracias!

pedrobaeza commented:
Como esto también está estancado mucho y no hay cambio, cierro el PR. Si sigue habiendo interés, reabridlo.
